### PR TITLE
Bar Chart for Ordinal Data, updates to Analyzer to be more dynamic (b…

### PR DESCRIPTION
**Pull request by laihipp**
…ased on info from file and not hardcoding column names)

### CODE REVIEW

File: Analyzer.py [hunk 1]
```diff
@@ -5,26 +5,30 @@
 
 
 
-def run_mean(pretest=None, posttest=None, ordinals=None, datatype="Interval"):
+def run_mean(data):
```
Comment (ian-mcnichols) [inline Analyzer.py:8]: If you're changing the parameters, change the docstrings

File: Analyzer.py [hunk 2]
```diff
@@ -5,26 +5,30 @@
...
+        # Run mean function on column data and add to results
+        results.append(np.mean(column))
+        max_column = i
+    # Find mean of difference between first set of data and last
+    difference = data[data.dtype.names[max_column]] - data[data.dtype.names[1]]
+    results.append(np.mean(difference))
+    return results
+
+
+def run_median(data, datatype="Interval"):
```
Comment (ian-mcnichols) [inline Analyzer.py:31]: Same issue, change docstrings

File: Analyzer.py [hunk 3]
```diff
@@ -166,25 +217,25 @@ def run_function(function_name, pretest=None, posttest=None, ordinals=None, data
     :return: function corresponding to operation type
     """
     if function_name == "Mean":
```
Comment (ian-mcnichols) [inline Analyzer.py:219]: Is this entire function just commented out?

File: Analyzer.py [hunk 4]
```diff
@@ -193,27 +244,27 @@ def run_function(function_name, pretest=None, posttest=None, ordinals=None, data
...
-        print("least squared line: ", run_least_square_line(my_data.data_np['Pretest'], my_data.data_np['Posttest']))
-        print("correlation coefficient:", run_correlation_coeff(my_data.data_np['Pretest'], my_data.data_np['Posttest']))
-        print("spearman coefficient: ",
-          run_spearman_rank_corr_coeff(my_data.data_np['Pretest'], my_data.data_np['Posttest']))
-    elif my_data.data_type == "Ordinal":
-        print("mean: ", run_mean(ordinals=my_data.ordinals, datatype=my_data.data_type))
-        print("median: ", run_median(ordinals=my_data.ordinals, datatype=my_data.data_type))
-        print("mode: ", run_mode(ordinals=my_data.ordinals, datatype=my_data.data_type))
-        print("probability dist: ", run_probability_dist(ordinals=my_data.ordinals, datatype=my_data.data_type))
+    if my_data.data_type == 'interval':
```
Comment (ian-mcnichols) [inline Analyzer.py:252]: What happened to all the other functions?

File: Analyzer.py [hunk 5]
```diff
@@ -193,27 +244,27 @@ def run_function(function_name, pretest=None, posttest=None, ordinals=None, data
...
+        # print("standard deviation: ", run_stand_dev(my_data.data_np))
+        # print("variance: ", run_variance(my_data.data_np)
+        # print("percentiles: ", run_percentiles(my_data.data_np)
+        # print("probability dist: ",
+        #   run_probability_dist(my_data.data_np['Pretest'], my_data.data_np['Posttest'], datatype=my_data.data_type))
+        # print("least squared line: ", run_least_square_line(my_data.data_np['Pretest'], my_data.data_np['Posttest']))
+        # print("correlation coefficient:", run_correlation_coeff(my_data.data_np['Pretest'], my_data.data_np['Posttest']))
+        # print("spearman coefficient: ",
+        #   run_spearman_rank_corr_coeff(my_data.data_np['Pretest'], my_data.data_np['Posttest']))
+    elif my_data.data_type == "ordinal":
```
Comment (ian-mcnichols) [inline Analyzer.py:267]: Are we only doing median and mode now?

File: Analyzer.py (hunk 1 above)
Comment (laihipp) [reply Analyzer.py:8]: Yea, I didn't get to that part, but I can do that tonight

**Review by ian-mcnichols — CHANGES_REQUESTED** [review]
This crashes when run from the GUI

**Pull request closed without merging**